platforms/allwinner-d1/boards/src/bin/mq-pro.rs

```diff
@@ -123,3 +128,63 @@ fn main() -> ! {
 
     d1.run()
 }
+
+fn init_i2c_puppet_irq(gpio: &mut d1_pac::GPIO, plic: &mut Plic) -> &'static WaitCell {
+    use mnemos_d1_core::plic::Priority;
+    use d1_pac::Interrupt;
+
+    static I2C_PUPPET_IRQ: WaitCell = WaitCell::new();
```

Comment:
Do you want to have some kind of singleton check here, to ensure this doesn't get called more than once?

Reply:
no, but we could add that, i suppose. i'm not sure if it matters, because i don't think it's a huge issue to set the same bits again...